### PR TITLE
Fix a couple of bugs in `Optimiser`

### PR DESCRIPTION
**Pull request by CoronelBuendia**
## Linked Issues

Errors could be thrown if there are no constraints (upon check  to see if the constraints are satisfied)

There was no way to add vector-valued equality constraints.
## Description

What is your PR trying to achieve? How did you go about achieving it?

## Interface Changes

Adds the previously non-existing `Optimiser.add_eq_constraints`
## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: bluemira/codes/_nlopt_api.py [hunk 1]
```diff
@@ -384,6 +384,26 @@ def add_eq_constraint(self, f_constraint, tolerance):
...
+            Constraint function
+        tolerance: float
+            Tolerance with which to enforce the constraint
+        """
+        if self.algorithm_name not in EQ_CON_ALGS:
+            raise OptUtilitiesError(
+                f"{self.algorithm_name} does not support equality constraints."
+            )
+
+        self._opt.add_equality_mconstraint(f_constraint, tolerance)
```
Comment (je-cook) [inline bluemira/codes/_nlopt_api.py:404]:
```suggestion
        self._opt.add_equality_constraint(f_constraint, tolerance)
```

**Review by je-cook — APPROVED** [review]
Just one typo (I think?)

**Review by hsaunders1904 — COMMENTED** [review]
Looks fine to me. Could do with a test for each of the bugs to guard against regressions?

File: bluemira/codes/_nlopt_api.py (hunk 1 above)
Comment (CoronelBuendia) [reply bluemira/codes/_nlopt_api.py:404]: nah this is right; its NLOPT for add matrix equality constraint

**Comment by je-cook** [comment]
@hsaunders1904 anything else to add?

**Pull request merged**